software/apps/cfft_radix4_q16/main.c

```diff
@@ -0,0 +1,214 @@
+                                      1);
+    mempool_bitrevtable_q16s_xpulpimg(
+        (uint16_t *)pSrc, BITREVINDEXTABLE_FIXED_TABLE_LENGTH, pRevT16);
+    pRes = pSrc;
+    mempool_stop_benchmark();
+  }
+  mempool_barrier(num_cores);
+#endif
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/* MULTI-CORE */
+#ifdef PARALLEL
+  int16_t *pRes; // Result pointer
+  mempool_start_benchmark();
+  mempool_radix4_cfft_q16p_xpulpimg(pSrc, (uint16_t)N_CSAMPLES, pCoef16_src, 1,
+                                    num_cores);
+  mempool_bitrevtable_q16p_xpulpimg((uint16_t *)pSrc,
+                                    BITREVINDEXTABLE_FIXED_TABLE_LENGTH,
+                                    pRevT16, num_cores);
+  pRes = pSrc;
+  mempool_stop_benchmark();
+#endif
+  mempool_barrier(num_cores);
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/* MULTI-CORE FOLDED */
+#ifdef FOLDED
+  int16_t *pRes; // Result pointer
+  if (core_id < (N_CSAMPLES / 16)) {
+    mempool_start_benchmark();
+#ifdef FOLDED_TWIDDLES
+    mempool_radix4_cfft_q16p_folded(pSrc, pDst, (uint16_t)N_CSAMPLES,
+                                    pCoef16_src, pCoef16_dst,
+                                    (N_CSAMPLES / 16));
+#else
+    mempool_radix4_cfft_q16p_folded(pSrc, pDst, (uint16_t)N_CSAMPLES,
+                                    pCoef16_src, (N_CSAMPLES / 16));
+#endif
+    pRes = ((LOG2 / 2) % 2) == 0 ? pSrc : pDst;
+    mempool_bitrevtable_q16p_xpulpimg((uint16_t *)pRes,
+                                      BITREVINDEXTABLE_FIXED_TABLE_LENGTH,
+                                      pRevT16, (N_CSAMPLES / 16));
+    mempool_stop_benchmark();
+  }
+  mempool_barrier(num_cores);
+#endif
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/* CHECK */
+#if defined(SINGLE) || defined(PARALLEL) || defined(FOLDED)
+  if (core_id == 0) {
+    printf("Done!\n");
+    for (uint32_t i = 0; i < N_RSAMPLES; i++) {
+      if (ABS(((int32_t)pRes[i] - (int32_t)vector_res[i])) > TOLERANCE)
+        printf("ERROR!!! Result[%d]: %6d Expected[%d]: %6d\n", i, pSrc[i], i,
+               vector_res[i]);
+    }
+  }
+  mempool_barrier(num_cores);
+#endif
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+/* MULTI-CORE SCHEDULED */
+#ifdef SCHEDULED
+  if (core_id < N_FFTs_COL * (N_CSAMPLES >> 4U)) {
+    mempool_start_benchmark();
+    uint32_t col_fftLen = N_CSAMPLES >> 2U;
+    uint32_t col_id = core_id / (N_CSAMPLES >> 4U);
+    mempool_radix4_cfft_q16p_scheduler(
+        pSrc, pDst, N_CSAMPLES, pCoef16_src + 2 * col_id * col_fftLen,
+        pCoef16_dst + 2 * col_id * col_fftLen, pRevT16,
+        BITREVINDEXTABLE_FIXED_TABLE_LENGTH, 1, N_CSAMPLES >> 4U);
+    mempool_log_partial_barrier(2, core_id, N_CSAMPLES >> 4U);
+    mempool_stop_benchmark();
+  }
+#endif
```

Comment:
What is the reason for executing this after the check? I.e., not checking the result of this implementation?

Reply:
Ok, I moved the check.